### PR TITLE
ts: add user types

### PR DESCRIPTION
**Pull request by sorja**
Part of #3

### CODE REVIEW

File: src/auth/index.ts [hunk 1]
```diff
@@ -1,3 +1,15 @@
-export interface AuthGroup {
+export interface Permissions {
+  // TODO
+}
+
+export interface RecordSteps {
   // TODO
 }
```
Comment (minotogna) [inline src/auth/index.ts:7]: add these now

File: src/index.ts [hunk 2]
```diff
@@ -1,8 +1,9 @@
+
 export type { AuthGroup } from './auth'
 export type { Cycle } from './cycle'
 export type { Labels } from './labels'
 export type { SRS } from './srs'
-export type { Validation } from './validation'
-
 export type { Survey } from './survey'
 export type { SurveyInfo } from './survey/info'
+export type { User } from './user'
```
Comment (minotogna) [inline src/index.ts:8]: put user under auth

File: src/user/index.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { Survey } from 'src/survey'
+import { AuthGroup } from 'src/auth';
+import { userStatus } from './userStatus';
+
+export interface Prefs {
```
Comment (minotogna) [inline src/user/index.ts:5]: call this UserPrefs

File: src/user/index.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  surveys: Array<Survey>
```
Comment (minotogna) [inline src/user/index.ts:6]: that's not an entire survey here... check the atomic props

File: src/user/userStatus.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export enum userStatus {
```
Comment (minotogna) [inline src/user/userStatus.ts:1]: upper case

**Review by minotogna — CHANGES_REQUESTED** [review]

File: src/auth/index.ts (hunk 1 above)
Comment (sorja) [reply src/auth/index.ts:7]: How can I populate data in these?

Comment (minotogna) [reply src/auth/index.ts:7]: check the code

File: src/user/userStatus.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
...
+  INVITED,
```
Comment (minotogna) [inline src/user/userStatus.ts:2]: the values are strings. not integers

File: src/auth/user/userStatus.ts [hunk 7]
```diff
@@ -0,0 +1,5 @@
+export enum UserStatus {
```
Comment (minotogna) [inline src/auth/user/userStatus.ts:1]: this can be moved in user file

File: src/auth/user/index.ts [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+import { AuthGroup } from 'src/auth'
+import { UserStatus } from './userStatus'
+
+export interface UserSurveys {
+  [key: number]: Survey
+  current: string
+}
+
+export interface UserPrefs {
+  surveys: Array<Survey>
```
Comment (minotogna) [inline src/auth/user/index.ts:11]: user doest not reference an entire survey. check my previous comment

File: src/auth/user/index.ts [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+export interface UserProps {
+  title: UserTitle
+}
+
+export interface User {
+  uuid: string
+  name: string
+  email: string
+  prefs: UserPrefs
+  props?: UserProps
```
Comment (minotogna) [inline src/auth/user/index.ts:32]: aren't these props required ?

Comment (sorja) [reply src/auth/user/index.ts:32]: They should, my newly created user had no props ({})

File: src/index.ts [hunk 10]
```diff
@@ -2,7 +2,7 @@ export type { AuthGroup } from './auth'
 export type { Cycle } from './cycle'
 export type { Labels } from './labels'
 export type { SRS } from './srs'
-export type { Validation } from './validation'
-
 export type { Survey } from './survey'
 export type { SurveyInfo } from './survey/info'
+export type { User } from './auth/user'
```
Comment (minotogna) [inline src/index.ts:7]: all user types and enum must be exported

File: src/record/recordStep.ts [hunk 11]
```diff
@@ -0,0 +1,16 @@
+export enum StepName {
+  entry = 'entry',
+  cleansing = 'cleansing',
+  analysis = 'analysis',
+}
+
+export type Step = {
```
Comment (minotogna) [inline src/record/recordStep.ts:7]: I would always use interface

File: src/record/recordStep.ts [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+  id: number
+  name: StepName
+}
+
```
Comment (minotogna) [inline src/record/recordStep.ts:11]: im not sure why we have Step and RecordStep.. ?

File: src/index.ts [hunk 13]
```diff
@@ -2,7 +2,7 @@ export type { AuthGroup } from './auth'
 export type { Cycle } from './cycle'
 export type { Labels } from './labels'
 export type { SRS } from './srs'
-export type { Validation } from './validation'
-
 export type { Survey } from './survey'
 export type { SurveyInfo } from './survey/info'
+export type { UserStatus, UserSurvey, UserSurveys, UserPrefs, UserTitle, UserProps, User } from './auth/user'
```
Comment (minotogna) [inline src/index.ts:7]: enum cannot be exported with export type

**Review by minotogna — APPROVED** [review]

**Pull request merged**